vavr/src/main/java/io/vavr/collection/Collections.java

```diff
@@ -330,6 +333,14 @@ public T next() {
         return scanLeft(reversedElements, zero, (u, t) -> operation.apply(t, u), us -> finisher.apply(reverseIterator(us)));
     }
 
+    static <T, U, R extends Seq<T>> R sortBy(Seq<? extends T> source, Comparator<? super U> comparator, Function<? super T, ? extends U> mapper, Collector<T, ?, R> collector) {
+        Objects.requireNonNull(comparator, "comparator is null");
+        Objects.requireNonNull(mapper, "mapper is null");
+        return source.toJavaStream()
+                .sorted((e1, e2) -> comparator.compare(mapper.apply(e1), mapper.apply(e2)))
+                .collect(collector);
```

Comment:
@ummels @ruslansennov it is not possible to keep the memoization. Because our equals() hashCode() methods rely on deep calculations. Even if we fix the sortBy case for `Vector<Stream<T>>`, we can't cover also ``Vector<Option<Stream<T>>>`` and so on. The only solution is to disable memoization.

However, memoization of collections may be not cheap at all, depending on the size of the collection. We do not cache the hashCode, currently it takes O(n) to calculate it - the same for equals().

Reply:
Thanks!